tidy3d/components/material/tcad/charge.py

```diff
@@ -300,7 +303,7 @@ class SemiconductorMedium(AbstractChargeMedium):
     )
 
     N_a: Union[pd.NonNegativeFloat, SpatialDataArray, tuple[DopingBoxType, ...]] = pd.Field(
```

Comment:
At some point we also discussed moving the `SpatialDataArray` to something like a `CustomDoping` doping box type. Do you foresee any issues or is it just a matter of implementing it? One usage difference is that then it will be additive with other doping boxes.